osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -154,10 +154,60 @@ protected override void LoadComplete()
 
             searchSection.Ruleset.BindValueChanged(_ => queueUpdateSearch());
             searchSection.Category.BindValueChanged(_ => queueUpdateSearch());
+            searchSection.Genre.BindValueChanged(_ => queueUpdateSearch());
+            searchSection.Language.BindValueChanged(_ => queueUpdateSearch());
             sortCriteria.BindValueChanged(_ => queueUpdateSearch());
             sortDirection.BindValueChanged(_ => queueUpdateSearch());
         }
 
+        public void ShowTag(string tag)
+        {
+            var currentQuery = searchSection.Query.Value;
+
+            if (currentQuery != tag)
+            {
+                setDefaultSearchValues();
+                searchSection.Query.Value = tag;
+            }
+
+            Show();
+        }
```

Comment:
I'm not sure what the point of these is either. This is another place that could be vastly simplified with the introduction of `SearchBeatmapSetsParameters`.